src/content_handlers/apt_handler/src/apt_handler.cpp

```diff
@@ -496,7 +505,12 @@ AptHandlerImpl::RemoveInstalledCriteria(const char* installedCriteriaFilePath, c
                         // Failed to remove.
                         success = false;
                     }
-                    break;
+
+                    if (!success)
+                    {
+                        break;
+                    }
+                    // otherwise, keep going to remove duplicates
                 }
```

Comment:
do we really want to serialize after every remove (line 500)?  seems like it would be better to move 500-501 outside of the loop.

Reply:
good point!  I will be sure to address this. I am closing this PR and will replicate to align with repo/branching strategy.